crates/stc_ts_type_checker/tests/conformance/controlFlow/controlFlowOptionalChain.error-diff.json

```diff
@@ -25,6 +25,8 @@
       223,
       241,
       244,
+      370,
```

Comment:
else, there should be separate logic in there, but putting it in false fact doesn't seem to work.
What am I missing?

Reply:
```ts
type Thing = { foo: string | number, bar(): number, baz: object };

function f16(o: Thing | undefined) {
    if (o?.foo === undefined) {
        o.foo;  // Error
    }
    else {
        o.foo;
    }
    if (o?.foo !== undefined) {
        o.foo;
    }
    else {
        o.foo;  // L370
    }
    if (o?.foo == undefined) {
        o.foo;  // Error
    }
    else {
        o.foo;
    }
    if (o?.foo != undefined) {
        o.foo;
    }
    else {
        o.foo;  // L382
    }
}
```